modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -395,7 +396,7 @@ private OutputMetadataDescriptor getOutputMetadataDescriptor()
      */
     private Object getKeyId(MetadataKey key) throws MetadataResolvingException
     {
-        return MetadataKeyIdObjectResolver.resolve(componentModel, key);
+        return new MetadataKeyIdObjectResolver().resolve(componentModel, key);
```

Comment:
don't instantiate one every time. Have one and keep it. If you do it like this you don't only add unnecessary pressure on the GC, you also loose the testability that we were trying to get.